src/common/lib/client/rest.ts

```diff
@@ -228,6 +230,155 @@ class Rest {
     }
   }
 
+  batchPublish(
+    specs: API.Types.BatchPublishSpec,
```

Comment:
Is it worth destructuring these types for the sake of brevity, ie:
```ts
type BatchPublishSpec = API.Types.BatchPublishSpec;
```

Reply:
Good idea, have updated.

Reply:
> declare an intermediate `BatchPublishResult` type

Cool, have done for presence and revocation too.

> use the `StandardCallback` type already available in this module (rather than `API.Types.StandardCallback`

The problem is that these two types are not equivalent. If you update the definition of `StandardCallback` in `src/common/types/utils.d.ts` to be `API.Types.StandardCallback` you'll see that it introduces a bunch of compilation errors. We have issue https://github.com/ably/ably-js/issues/1405 for fixing this. That is, we should be using `API.Types.StandardCallback` throughout the codebase to ensure we're conforming to the public API, and I think it hence makes sense to use it for any new code.

Reply:
ah ok that makes sense, thanks for explaining!